src/serac/physics/utilities/variational_form/integral.hpp

```diff
@@ -74,19 +114,26 @@ void Add(const mfem::DeviceTensor<3, double>& r_global, tensor<double, ndof, com
   }
 }
 
-// the impl::Preprocess computes the arguments to be passed into the q-function
-//
-// by default:
-//   H1 family elements will compute {value, gradient}
-//   Hcurl family elements will compute {value, curl}
-//   Hdiv family elements will compute {value, divergence}  TODO
-//   L2 family elements will compute value                  TODO
-//
-// in the future, the user will be able to override these defaults
-// to omit unused components (e.g. specify that they only need the gradient)
+/**
+ * @brief Computes the arguments to be passed into the q-function (shape function evaluations)
+ * By default:
+ *  H1 family elements will compute {value, gradient}
+ *  Hcurl family elements will compute {value, curl}
+ *  TODO: Hdiv family elements will compute {value, divergence}
+ *  TODO: L2 family elements will compute value
+ *
+ * In the future, the user will be able to override these defaults
+ * to omit unused components (e.g. specify that they only need the gradient)
+ *
+ * @param[in] u The DOF values for the element
+ * @param[in] xi The position of the quadrature point in reference space
+ * @param[in] J The Jacobian of the element transformation at the quadrature point
+ * @tparam element_type The type of the element (used to determine the family)
+ */
 template <typename element_type, typename T, int dim>
 auto Preprocess(T u, const tensor<double, dim> xi, const tensor<double, dim, dim> J)
 {
+  // TODO: Static assert that element_type is an element??
```

Comment:
maybe something like this
```suggestion
static_assert(is_finite_element<element_type>::value, "impl::Preprocess error: template argument must be of type `finite_element<>`);
```